hare/eligibility/oracle.go

```diff
@@ -159,3 +204,52 @@ func (o *Oracle) Proof(id types.NodeId, layer types.LayerID, round int32) ([]byt
 
 	return sig, nil
 }
+
+// Returns a map of all active nodes in the specified layer id
+func (o *Oracle) actives(layer types.LayerID) (map[string]struct{}, error) {
+	sl := roundedSafeLayer(layer, k, o.layersPerEpoch, constOffset)
+
+	ep := sl.GetEpoch(o.layersPerEpoch)
+	// check genesis
+	if ep.IsGenesis() {
+		return nil, genesisErr
```

Comment:
why not just return genesisActiveSet?

Reply:
actives just calculates/returns the active set. He doesn't know what is the meaning of genesisErr for the caller. For example, IsIdentityActiveOnConsensusView returns true for everyone if genesisErr is returned.